src/main/webapp/script.js

```diff
@@ -12,9 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 var arrayOfProducts = Array();
 
 async function fetchDataAsync() {
     const response = await fetch('test.html');
     arrayOfProducts = await response.json();
+}
+
+function filtering(){
+    var checkBox = document.getElementsBy("result");
+    if(checkBox.checked == true){
+        for(i = 0; i <  arrayOfProducts.length; i++){
+           let prodbrand = arrayOfProducts[i].brand;
```

Comment:
variable name seems wrong... prodbrand here but prodBrand in line 29?

Reply:
fixed